SayTheirNames/Source/Dependency/DependencyContainer.swift

```diff
@@ -24,23 +24,63 @@
 
 import UIKit
 
-protocol Servicing {
-    var image: ImageService { get }
-    var dateFormatter: DateFormatterService { get }
-    var network: NetworkRequestor { get }
+// MARK: - Dependency
+protocol Dependency {
+    
 }
-/// This is a core class that holds all instances responsible for logic
-final class Service: Servicing {
+
+// MARK: - DependencyContainer
+final class DependencyContainer: Dependency {
+    lazy private(set) var navigator = Navigator()
     lazy private(set) var image = ImageService()
     lazy private(set) var dateFormatter = DateFormatterService()
     lazy private(set) var network = NetworkRequestor()
     
-    static let shared = Service()
-    
     // MARK: - Init
     init() {
         Log.mode = .all
         Log.print("SayTheirNames Version: \(Bundle.versionBuildString)")
         Log.print("Starting Services")
+        
+        // Handle our Injection
+        let injectionFactory = DependencyInjectionFactory.notShared
+        injectionFactory.add(handle: self)
+        injectionFactory.add(handle: self.navigator)
+        injectionFactory.add(handle: self.image)
+        injectionFactory.add(handle: self.dateFormatter)
+        injectionFactory.add(handle: self.network)
+    }
+}
+
+// MARK: - ServiceInject
+@propertyWrapper
+struct DependencyInject<S: Dependency> {
+    var serviceHandle: S
+    
+    init() {
+        self.serviceHandle = DependencyInjectionFactory.notShared.resolve(S.self)
+    }
+    
+    public var wrappedValue: S {
+        get { serviceHandle }
+        mutating set { serviceHandle = newValue }
+    }
+}
+
+// MARK: - InjectionFactory
+private class DependencyInjectionFactory {
+    static let notShared = DependencyInjectionFactory()
+    var factoryDict: [String: Any] = [:]
+    
+    func add<S: Dependency>(handle: S) {
+        let key = String(describing: handle.self)
+        self.factoryDict[key] = handle
+    }
+
+    func resolve<S: Dependency>(_ type: S.Type) -> S {
+        let key = String(reflecting: type)
+        let component = self.factoryDict[key]
+        
+        return component as! S
```

Comment:
guard & fatal error would be better here (minor thing though)

Reply:
Added